tests/kit/src/utils/keyboard.ts

```diff
@@ -102,25 +104,42 @@ export async function undoByKeyboard(page: Page) {
   await keyUpCtrlOrMeta(page);
 }
 
-export async function writeTextToClipboard(page: Page, text: string) {
+const clipboardMutex = {
+  lock: Promise.resolve(),
+  async acquire() {
+    await this.lock;
+    const { promise, resolve } = Promise.withResolvers<void>();
+    this.lock = promise;
+    return resolve;
+  },
+};
+
```

Comment:
The lock may not function correctly if two parties (B/C) are waiting for the same lock (created by A). After A released, both B and C will create their own new lock at line 112.

btw you can use `AsyncLock` from `@toeverything/infra`